relay-server/src/processing/transactions/mod.rs

```diff
@@ -0,0 +1,592 @@
+
+use crate::Envelope;
+use crate::envelope::{ContentType, EnvelopeHeaders, Item, ItemType};
+use crate::managed::{
+    Counted, Managed, ManagedEnvelope, ManagedResult, OutcomeError, Quantities, Rejected,
+};
+use crate::processing::transactions::profile::Profile;
+use crate::processing::utils::event::{
+    EventFullyNormalized, EventMetricsExtracted, FiltersStatus, SpansExtracted, event_type,
+};
+use crate::processing::{Forward, Processor, QuotaRateLimiter, RateLimited, utils};
+use crate::services::outcome::{DiscardReason, Outcome};
+use crate::services::processor::{ProcessingError, ProcessingExtractedMetrics};
+#[cfg(feature = "processing")]
+use crate::services::store::StoreEnvelope;
+use crate::statsd::{RelayCounters, RelayTimers};
+use crate::utils::{SamplingResult, should_filter};
+
+#[cfg(feature = "processing")]
+use crate::managed::TypedEnvelope;
+#[cfg(feature = "processing")]
+use crate::services::processor::ProcessingGroup;
+
+pub mod extraction;
+mod process;
+pub mod profile;
+pub mod spans;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("invalid JSON")]
+    InvalidJson(#[from] serde_json::Error),
+    #[error("envelope processor failed")]
+    ProcessingFailed(#[from] ProcessingError),
+    #[error("rate limited")]
+    RateLimited(RateLimits),
+}
+
+impl From<RateLimits> for Error {
+    fn from(value: RateLimits) -> Self {
+        Self::RateLimited(value)
+    }
+}
+
+impl OutcomeError for Error {
+    type Error = Self;
+
+    fn consume(self) -> (Option<Outcome>, Self::Error) {
+        let outcome = match &self {
+            Self::InvalidJson(_) => Outcome::Invalid(DiscardReason::InvalidJson),
+            Self::ProcessingFailed(e) => match e {
+                ProcessingError::InvalidTransaction => {
+                    Outcome::Invalid(DiscardReason::InvalidTransaction)
+                }
+                ProcessingError::EventFiltered(key) => Outcome::Filtered(key.clone()),
+                _other => {
+                    relay_log::error!(
+                        error = &self as &dyn std::error::Error,
+                        "internal error: trace metric processing failed"
+                    );
+                    Outcome::Invalid(DiscardReason::Internal)
+                }
+            },
+            Self::RateLimited(limits) => {
+                let reason_code = limits.longest().and_then(|limit| limit.reason_code.clone());
+                Outcome::RateLimited(reason_code)
+            }
+        };
+        (Some(outcome), self)
+    }
+}
+
+/// A processor for transactions.
+pub struct TransactionProcessor {
+    limiter: Arc<QuotaRateLimiter>,
+    geoip_lookup: GeoIpLookup,
+    #[cfg(feature = "processing")]
+    quotas_client: Option<AsyncRedisClient>,
+    reservoir_counters: ReservoirCounters,
+}
+
+impl Processor for TransactionProcessor {
+    type UnitOfWork = SerializedTransaction;
+
+    type Output = TransactionOutput;
+
+    type Error = Error;
```

Comment:
```suggestion
    type UnitOfWork = SerializedTransaction;
    type Output = TransactionOutput;
    type Error = Error;
```

Baby nit, that's how the other processors are formatted.